--- a/Source/GUI/Qt/main.cpp
+++ b/Source/GUI/Qt/main.cpp
@@ -32,6 +32,12 @@ int main(int argc, char *argv[])
     QApplication a(argc, argv);
     QStringList args = QCoreApplication::arguments();
 
+    QString progName = QFileInfo(a.applicationFilePath()).fileName();
+    if (progName == QFileInfo(args.at(0)).fileName()) {
+        // First argument is the program name.
+        args.removeFirst();
+    }
+
     foreach(QString arg,args) {
         if(!arg.compare("--help",Qt::CaseInsensitive)||!arg.compare("-h",Qt::CaseInsensitive)) {
             cout << "Usage : mediainfo-gui [OPTIONS] [files]" << endl;
--- a/Source/GUI/Qt/mainwindow.cpp
+++ b/Source/GUI/Qt/mainwindow.cpp
@@ -222,7 +222,7 @@ MainWindow::MainWindow(QStringList filesnames, int viewasked, QWidget *parent) :
 
     refreshDisplay();
 
-    if(filesnames.count()>1) {
+    if(filesnames.count()>0) {
         openFiles(filesnames);
     }
 
